### PR TITLE
[tests] Clean up retention SRAM E2E tests

### PR DESCRIPTION
**Pull request by milesdai**
Signed-off-by: Miles Dai <milesdai@google.com>

### CODE REVIEW

File: sw/device/silicon_creator/rom/e2e/rom_e2e_ret_ram_init_test.c [hunk 1]
```diff
@@ -42,15 +41,18 @@ rom_error_t retention_ram_init_test(void) {
...
-    memcpy(raw, ret, sizeof(retention_sram_t));
 
+    static_assert(sizeof(retention_sram_t) % sizeof(uint64_t) == 0,
+                  "This test expects the retention SRAM size to be a multiple "
+                  "of uint64_t");
     uint32_t matches = 0;
-    for (size_t i = 0; i < ARRAYSIZE(raw); ++i) {
-      if (raw[i] == pattern64) {
+    for (size_t i = 0; i < sizeof(retention_sram_t) / sizeof(uint64_t); i++) {
+      if (read_64((char *)ret + i * sizeof(uint64_t)) == pattern64) {
```
Comment (alphan) [inline sw/device/silicon_creator/rom/e2e/rom_e2e_ret_ram_init_test.c:50]: nit, just to make things simpler:
```suggestion
    for (size_t i = 0; i < sizeof(retention_sram_t); i += sizeof(uint64_t)) {
      if (read_64((char *)ret + i) == pattern64) {
```

File: sw/device/silicon_creator/rom/e2e/rom_e2e_ret_ram_keep_test.c [hunk 2]
```diff
@@ -30,20 +30,19 @@ enum {
...
-  memcpy(raw, ret, sizeof(retention_sram_t));
-
   bool unchanged = true;
-  // Skip checking index 0 since that is used to store the
+  // Skip checking word 0 since that is used to store the
   // reset reason and will be changed.
-  for (size_t i = 1; i < ARRAYSIZE(raw); ++i) {
-    if (raw[i] != pattern32) {
+  for (size_t i = 1; i < sizeof(retention_sram_t) / sizeof(uint32_t); ++i) {
+    uint32_t val = read_32((char *)ret + i * sizeof(uint32_t));
```
Comment (alphan) [inline sw/device/silicon_creator/rom/e2e/rom_e2e_ret_ram_keep_test.c:42]: nit, same as above:
```suggestion
  for (size_t i = 1; i < sizeof(retention_sram_t); i += sizeof(uint32_t)) {
    uint32_t val = read_32((char *)ret + i);
```

**Review by alphan — APPROVED** [review]

File: sw/device/silicon_creator/rom/e2e/rom_e2e_ret_ram_init_test.c (hunk 1 above)
Comment (milesdai) [reply sw/device/silicon_creator/rom/e2e/rom_e2e_ret_ram_init_test.c]: Done

File: sw/device/silicon_creator/rom/e2e/rom_e2e_ret_ram_keep_test.c (hunk 2 above)
Comment (milesdai) [reply sw/device/silicon_creator/rom/e2e/rom_e2e_ret_ram_keep_test.c]: Done with modifications:

The starting index for `i` is now `sizeof(uint32_t)`

**Pull request merged**